WooCommerce/Classes/AppDelegate.swift

```diff
@@ -191,7 +194,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
             // When the app is put into an incative state, it's important to ensure that any pending changes to
             // Core Data context are saved to avoid data loss.
-            ServiceLocator.storageManager.viewStorage.saveIfNeeded()
+            storageManager.viewStorage.saveIfNeeded()
```

Comment:
I am trying to understand whether we still need to save the `viewStorage` during app transitions. 

As per my understanding,

- We always perform the write operations to the `writerDerivedStorage`. 
- The `viewContext` will automatically receive these changes as we have set `automaticallyMergesChangesFromParent` as true on `viewContext`. 
- The `viewContext` should never have changes that are different from the `persistentContainer` that needs to be saved before quitting the app. 

What do you think? 

Tagging @pmusolino as he authored the changes [here](https://github.com/woocommerce/woocommerce-ios/pull/13247)

Reply:
As mentioned in the reference of my PR peaMlT-Jf#comment-1773, this is used also in example provided by Apple. There does not have to be changes, so it is just an extra check to be sure, in fact it will only be saved if there really are changes, otherwise it will do nothing.

Reply:
Thanks, @pmusolino! 

As I explained earlier, I think the `viewContext` will not have any changes to save, and I propose removing this code. If you and @itsmeichigo agree, we can add a subtask and deal with it as part of the core data usage update project.

Reply:
I agree, and I can remove these lines in this PR directly. `saveIfNeeded` has been marked as deprecated already and we should be saving changes in the background context right after the write operations.

Reply:
Done in e6e644d9d5f50c092f73b351d9361b812faa20ea.